ports/stm32/mboot/main.c

```diff
@@ -698,9 +700,18 @@ static int flash_write(uint32_t addr, const uint8_t *src8, size_t len) {
 /******************************************************************************/
 // Writable address space interface
 
-static int do_mass_erase(void) {
-    // TODO
-    return flash_mass_erase();
+static int do_mass_erase() {
+    uint32_t sector_num = 0;
+    for (int i = 0; i < MP_ARRAY_SIZE(flash_layout); ++i) {
+        for (int j = 0; j < flash_layout[i].sector_count; ++j) {
+            sector_num ++;
+        }
+    }
+    if (flash_erase_hal(1, sector_num-1) != 0) {
```

Comment:
note that some MCUs (eg WB55) have mboot in multiple sectors because the sector size is 4k, so starting at sector 1 won't work for them.  well, I think that's a bit out of scope of this PR, it can be fixed later (so this mass erase only works on F4/F7s for the time being)